client.js

```diff
@@ -90,6 +116,18 @@ RingpopClient.prototype._request = function _request(host, endpoint, head, body,
             return;
         }
 
+        if (opts.gzip === true) {
```

Comment:
It would make more sense to me if the headers of the response would indicate that the body is gzipped.

With the current PR both nodes involved in a sync would need to have the same configuration which could cause strange issues in production one day. By indicating gzippedness in the header of the response the client can deduct on its own that it needs to unzip the contents before parsing it.